sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientConfigurationTests.java

```diff
@@ -141,4 +151,105 @@ void testResourceServerWithOboExistCustomAndGraphClient() {
                 assertThat(customScopes).containsOnly("api://52261059-e515-488e-84fd-a09a3f372814/File.Read");
             });
     }
+
+    @Test
+    void testJwtBearerOAuth2AuthorizedClientProviderAuthExceptionWhenUsingPrivateKeyJwtMethod() {
+        OAuth2ClientAuthenticationJwkResolver jwkResolver = spy(new TestOAuth2ClientAuthenticationJwkResolver());
+        resourceServerWithOboContextRunner()
+            .withBean(OAuth2ClientAuthenticationJwkResolver.class, () -> jwkResolver)
+            .withUserConfiguration(AadOAuth2ClientConfiguration.class)
+            .withPropertyValues(
+                "spring.cloud.azure.active-directory.enabled=true",
+                "spring.cloud.azure.active-directory.credential.client-certificate-path=/test/test.pfx",
+                "spring.cloud.azure.active-directory.credential.client-certificate-password=test",
```

Comment:
the certificate is not used if `TestOAuth2ClientAuthenticationJwkResolver` is used.

Reply:
Yes, I assume here is testing the bean creation , the `OAuth2ClientAuthenticationJwkResolver` test case is this class [AadOAuth2ClientAuthenticationJwkResolverTests](https://github.com/moarychan/azure-sdk-for-java/blob/316164cf154ce24b317d92b02151783b66fcef00/sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientAuthenticationJwkResolverTests.java#L28)